src/dist/replication/meta_server/duplication/duplication_info.cpp

```diff
@@ -191,5 +191,25 @@ duplication_info_s_ptr duplication_info::decode_from_blob(dupid_t dup_id,
     return dup;
 }
 
+void duplication_info::append_if_valid_for_query(
+    const app_state &app,
+    /*out*/ std::vector<duplication_entry> &entry_list) const
+{
+    zauto_read_lock l(_lock);
+
+    entry_list.emplace_back(to_duplication_entry());
+    duplication_entry &ent = entry_list.back();
+    ent.__isset.not_confirmed = true;
+    // the confirmed decree is not useful for displaying
+    // the overall state of duplication, instead we show pending mutations.
+    ent.__isset.progress = false;
+    for (const partition_configuration &part : app.partitions) {
+        int pid = part.pid.get_partition_index();
+        auto it = _progress.find(pid);
```

Comment:
Does each pid has a corresponding entry in _progress?

Reply:
Yes, `progress` is used to sync duplication confirmed decree between meta and replica. Each partition within the duplicating app reports its confirmed decree to meta, so that during failure recovery, the partition can restore the former progress and restart duplication.